### ISSUE TITLE
json_encode(): Invalid UTF-8 sequence

### ISSUE THREAD

**Issue by tobiasschaeferptb** [fabarea/media#143]
If a user is working with the media module with files with special characters (e.g. "é") in the title this error occurs in the TYPO3 log :
Core: Error handler (BE): PHP Warning: json_encode(): Invalid UTF-8 sequence in argument in typo3conf/ext/vidi/Classes/ViewHelpers/Result/ToJsonViewHelper.php line 43

The title of those files isn't shown in the list.
Since json_encode can only handle UTF-8 a conversion is needed.
Something like this:
    public function render() {

```
    $objects = $this->templateVariableContainer->get('objects');
    $columns = $this->templateVariableContainer->get('columns');
    $output = array(
        'sEcho' => $this->getNextTransactionId(),
        'iTotalRecords' => $this->templateVariableContainer->get('numberOfObjects'),
        'iTotalDisplayRecords' => $this->templateVariableContainer->get('numberOfObjects'),
        'iNumberOfRecords' => count($objects),
        'aaData' => $this->getRowsViewHelper()->render($objects, $columns),
    );

    // T. Schaefer, 2016-07-27: added call of encodeItems
    $output = $this->encodeItems($output);
    $this->setHttpHeaders();
    return json_encode($output);
}
```

...

```
/** T. Schaefer, 2016-07-27: encodeItems converts the values of an array to UTF-8
 * @return array
 */
protected function encodeItems($array) {
    foreach($array as $key => $value) {
        if(is_array($value)) {
            $array[$key] = $this->encodeItems($value);
        }
        else {
            $array[$key] = utf8_encode($value);
        }
    }
    return $array;
}
```

**Comment by tobiasschaeferptb** [comment]
I forgot to mention:
TYPO3 6.2.26
Media: 3.7.6
Vidi: 1.3.1

Since the coding in vidi/Classes/ViewHelpers/Result/ToJsonViewHelper.php in version 2.2.0 is almost the same like in 1.3.1 I assume the error still occurs with media 4.1.0 & vidi 2.2.0.

**Comment by fabarea** [comment]
Will be fixed in EXT:vidi

**Comment by fabarea** [comment]
I had to revert the patch. It was making a double utf-8 encoding on already utf-8 encoded string for me. Could you try the function below and let me know if it does the job.

```

    /**
     * @param array $values
     * @return mixed
     */
    protected function encodeItems(array $values)
    {
        foreach ($values as $key => $value) {
            if (is_array($value)) {
                $values[$key] = $this->encodeItems($value);
            } elseif (mb_detect_encoding($value, 'UTF-8', true) !== 'UTF-8') {
                $values[$key] = utf8_encode($value);
            }
        }
        return $values;
    }
```

**Comment by tobiasschaeferptb** [comment]
I tried your coding and the "Invalid UTF-8 sequence in argument" error doesn't occur. But titles with special characters are displayed wrong in the media module, e.g. "RÃ©fÃ©rences" instead of "Références". Using phpmyadmin the title looks to be stored right in the database. This behaviour was the some with the encodeItems function I suggested. Unfortunately I couldn't find a solution for displaying the special characters correctly in the media module.
